src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -34,29 +34,32 @@
 
     Optional<User> findOneByLogin(String login);
 
-    @EntityGraph(attributePaths = { "authorities", "groups" })
-    Optional<User> findOneWithGroupsAndAuthoritiesByLogin(String login);
+    @Query("select distinct user from User user left join fetch user.groups left join fetch user.authorities where user.login = :#{#login}")
+    Optional<User> findOneWithGroupsAndAuthoritiesByLogin(@Param("login") String login);
```

Comment:
What is the reason that for the methods in the repository now explicit queries are used?
It seems to me that they have the same outcome.

Reply:
I am not quite sure, but it seemed that @EntityGraph does `outer join` while an explicit `left join` can be way more efficient in the database, see e.g. https://www.quora.com/What-is-the-difference-between-left-join-and-left-outer-join-in-sql. 

Unfortunately I could not find an exact definition, how EntityGraph is doing the join.

Reply:
> Wow, there was a lot of unnecessary database communication that you cleaned up here 👍
> I didn't find anything major and also didn't see any bugs when I was running this branch.
> 
> Two client tests are currently failing for this branch, that are not working locally for on other branches 🤔
> @lovaiible Could you check those? https://bamboobruegge.in.tum.de/build/result/viewBuildResults.action?buildKey=ARTEMIS-TESTS18-TSTEST&buildNumber=6

I already asked @lovaiible to work on this. She is currently doing it

Reply:
We should probably find this out because it might have performance impact in other places. If they do the same, we can basically change back. Otherwise I would prefer the explicit query with `left join fetch`

Reply:
`@EntityGraph` does a `LEFT OUTER JOIN` and there is now way of changing that behavior.
It is possible to use `@NamedEntityGraph` to specify more detailed what to load,  but if I understand correctly, this still uses an `OUTER JOIN`. If we want to do a `LEFT JOIN` or even a `LEFT INNER JOIN` we need to use JPA Queries.

Reply:
Good to know!